docroot/modules/humsci/hs_layouts/hs_layouts.module

```diff
@@ -182,9 +182,10 @@ function _hs_layouts_make_title_the_link(&$variables) {
       $variables['href'] = $variables['href'] ?? htmlspecialchars_decode(trim(strip_tags($renderer->renderInIsolation($variables['button']))));
     }
   }
+
   if (!empty($variables['title'])) {
     $title = is_array($variables['title']) ? $renderer->renderInIsolation($variables['title']) : $variables['title'];
-    $variables['title'] = ['#markup' => trim(strip_tags($title))];
+    $variables['title'] = $title ?? ['#markup' => trim(strip_tags($title))];
```

Comment:
Why does this work? I'm not really understanding.

if `$variables['title']` is a render array, we get the rendering, otherwise it's would be a string. But either way `$title` is not null/false, so it would always set the value to what is returned in line 187. The `?? ....` doesn't really do anything at this point.

Reply:
The thing is that in line 187 `$title` returned an empty string. So in line 188, if we don't add the `??` it will set $variables['title'] like this:

![Screenshot 2025-01-09 at 8 42 55 AM](https://github.com/user-attachments/assets/0aaa31f5-11da-4cc6-afa9-f1a01526c299)

$variables['title'] will return an array in these scenario, so in the template `vertical-card.html.twig` it will seem that there's something in `title` even though it's an array with an `#markup` that is empty.

Reply:
I'm also trying to understand what's going on here. If `$variables['title']` is equal to an empty string it shouldn't get past the `empty()` conditional on line 186. Unless the `renderInIsolation()` method is returning an empty string?

Additionally, on line 188 if `$title` is empty, what's the purpose of stripping tags from an empty string?